xbmc/music/windows/GUIWindowMusicBase.cpp

```diff
@@ -690,12 +692,18 @@ bool CGUIWindowMusicBase::OnContextButton(int itemNumber, CONTEXT_BUTTON button)
       Update(item->GetPath());
       int bookmark;
       m_musicdatabase.GetResumeBookmarkForAudioBook(item->GetPath(), bookmark);
-      int i=0;
-      while (i < m_vecItems->Size() && bookmark > m_vecItems->Get(i)->m_lEndOffset)
-        ++i;
-      CFileItem resItem(*m_vecItems->Get(i));
-      resItem.SetProperty("StartPercent", ((double)bookmark-resItem.m_lStartOffset)/(resItem.m_lEndOffset-resItem.m_lStartOffset)*100);
-      g_application.PlayFile(resItem, "", false);
+
+      auto item_it = std::find_if(
+        m_vecItems->begin(),
+        m_vecItems->end(),
+        [&](const CFileItemPtr& item) { return bookmark <= item->m_lEndOffset; }
+      );
+      if(item_it != m_vecItems->end())
+      {
+        CFileItem resItem(**item_it);
+        resItem.SetProperty("StartPercent", ((double)bookmark-resItem.m_lStartOffset)/(resItem.m_lEndOffset-resItem.m_lStartOffset)*100);
```

Comment:
```suggestion
        resItem.SetProperty("StartPercent", (static_cast<double> (bookmark-resItem.m_lStartOffset) / (resItem.m_lEndOffset - resItem.m_lStartOffset) * 100);
```

code style

Reply:
That's @notspiff 's fault :stuck_out_tongue: but I will fix it

Reply:
Agreed. Just realized the cast is not even necessary if we replace `100` with `100.0`, which will trigger implicit promotion to double.